### PR TITLE
User Profile - capture domain when creating API keys and tokens

### PR DESCRIPTION
**Pull request by ywangd**
Creation of API keys and tokens already capture realm information about
the user. This PR adds to the process so that domain information is also
captured when domain is configured.

Note that we decide to only capture the domain info, but NOT use it for
the time being. It may be useful in future handling of domain spliting.
But it is not yet decided. Consequently, we deliberately avoid exposing
this info to any consumers, e.g. API key owner's domain is not added to
the authentication metadata.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java [hunk 1]
```diff
@@ -1798,15 +1798,20 @@ static BytesReference createTokenDocument(
...
                     .field("type", "unassociated_client")
                     .field("user", originatingClientAuth.getUser().principal())
-                    .field("realm", originatingClientAuth.getAuthenticatedBy().getName())
-                    .endObject()
-                    .endObject();
+                    .field("realm", originatingClientAuth.getAuthenticatedBy().getName());
+                if (originatingClientAuth.getSourceRealm().getDomain() != null) {
+                    builder.field("realm_domain", originatingClientAuth.getSourceRealm().getDomain());
+                }
+                builder.endObject().endObject();
```
Comment (ywangd) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java:1805]: Existing code capture the `authenticatedBy` realm and I think it's wrong. I have separate PR (#84263) for fix it.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Comment by ywangd** [comment]
@elasticmachine update branch

**Pull request merged**